security/pkg/pki/util/generate_cert.go

```diff
@@ -81,19 +82,35 @@ type CertOptions struct {
 
 	// If true, the private key is encoded with PKCS#8.
 	PKCS8Key bool
+
+	// Whether this certificate uses Elliptical Curve. Mutually
+	// exclusive with respect to RSAKeySize.
+	IsEC bool
 }
 
 // GenCertKeyFromOptions generates a X.509 certificate and a private key with the given options.
 func GenCertKeyFromOptions(options CertOptions) (pemCert []byte, pemKey []byte, err error) {
-	// Generate a RSA private&public key pair.
+	// Generate the appropriate private&public key pair based on options.
 	// The public key will be bound to the certificate generated below. The
 	// private key will be used to sign this certificate in the self-signed
 	// case, otherwise the certificate is signed by the signer private key
 	// as specified in the CertOptions.
-	priv, err := rsa.GenerateKey(rand.Reader, options.RSAKeySize)
+	if options.IsEC {
+		ecPriv, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+		if err != nil {
+			return nil, nil, fmt.Errorf("cert generation fails at EC key generation (%v)", err)
+		}
+		return genCert(options, ecPriv, &ecPriv.PublicKey)
```

Comment:
nit: can you merge this block with the code below?

Reply:
Done